### PR TITLE
refine patch out semaphore compat

### PR DESCRIPTION
**Pull request by shayne-fletcher**
more changes to 'compiler/driver/' [^1] requires we refine the patching first introduced in https://github.com/digital-asset/ghc-lib/pull/447 a bit

[^1]: e.g. https://gitlab.haskell.org/ghc/ghc/-/commit/86aae5702d09db2f50c42a3f43ef72df1e3a710b https://gitlab.haskell.org/ghc/ghc/-/commit/87eebf98cb485f7c9175330051736e147ade9848 https://gitlab.haskell.org/ghc/ghc/-/commit/4d29ecdfcc79ad663e066d9f7d6d17b64c8c6c41

### CODE REVIEW

**Comment by shayne-fletcher** [comment]
cc @moisesackerman-da @garyverhaegen-da @dylant-da

**Comment by basvangijzel-DA** [comment]
FYI, it might take a few days to get reviewed as most of the team is out this week.

**Comment by shayne-fletcher** [comment]
thanks for letting me know @basvangijzel-DA !

File: patches/2972fd66f91cb51426a1df86b8166a067015e231-GHC_Driver_Make_hs.patch
Comment (akrmn) [inline patches/2972fd66f91cb51426a1df86b8166a067015e231-GHC_Driver_Make_hs.patch:1]: it seems the changes in this patch only change whitespace, so I don't see why we need it?

Comment (shayne-fletcher) [reply patches/2972fd66f91cb51426a1df86b8166a067015e231-GHC_Driver_Make_hs.patch:1]: @moisesackerman-da the diff is showing changes between the old patch and the new patch. that's indeed due to the whitespace changes introduced in https://gitlab.haskell.org/ghc/ghc/-/commit/4d29ecdfcc79ad663e066d9f7d6d17b64c8c6c41. with these whitespace changes the existing patch won't apply. redoing the patch with these changes in whitespace accounted for fixes the problem.

Comment (shayne-fletcher) [reply patches/2972fd66f91cb51426a1df86b8166a067015e231-GHC_Driver_Make_hs.patch:1]: i did get the name of the replacement patch wrong though. should be `4d29ecdfcc79ad663e066d9f7d6d17b64c8c6c41-GHC_Driver_Make_hs.patch` which i've now pushed a correction for. for the avoidance of doubt, it is precisely the patch file obtained by the command `git diff 4d29ecdfcc79ad663e066d9f7d6d17b64c8c6c41..5c8731244bc13a3d813d2a4d53b3188b28dc8355^1 -- compiler/GHC/Driver/Make.hs`.

**Comment by shayne-fletcher** [comment]
> LGTM, just a question about the new `GHC_Driver_Make_hs.patch` file

alright @akrmn? any more questions?

**Pull request merged**